lib/beaker/hypervisor/docker.rb

```diff
@@ -446,12 +445,19 @@ def fix_ssh(container, host=nil)
     # return the existing container if we're not provisioning
     # and docker_container_name is set
     def find_container(host)
-      return nil if host['docker_container_name'].nil? || @options[:provision]
-      @logger.debug("Looking for an existing container called #{host['docker_container_name']}")
-
-      ::Docker::Container.all.select do |c|
-        c.info['Names'].include? "/#{host['docker_container_name']}"
-      end.first
+      if (id = host['docker_container_id'])
```

Comment:
how does this take care of the "not provisioning" case?

Reply:
I didn't think it made sense to check for `:provision` in this method, since the name `find_container` suggests that the only purpose is only to find the container -- my thinking is that the check should happen separately. I put this check further up, [here](https://github.com/puppetlabs/beaker-docker/pull/13/files#diff-d7fdc977ad52b213077190f7ec17f42fR130) instead

Reply:
👍